Grant: SubSMT

YanOctavian opened the pull request:
#### Project Abstract

On-chain resources are precious, and more and more teams, especially L2, 
need to put data and calculations off-chain, and Merkel trees are used more frequently.
Sparse Merkle trees are widely used in blockchains, and have been used hundreds of thousands of times in the Ethereum community. 
Polkadot uses the rust language as its development language, and the implementation of sparse Merkel trees currently lacks a unified solution. 
Especially in substrate and ink, if there is an implementation template for sparse Merkle trees, 
it will greatly save the community's development time and avoid repeated development. 
This project will do just that.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [x] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ Polkadot (USDC & USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

octavei commented:
> I have read and hereby sign the Contributor License Agreement.

I have read and hereby sign the Contributor License Agreement.

YanOctavian commented:
> I have read and hereby sign the Contributor License Agreement.

I have read and hereby sign the Contributor License Agreement.

semuelle commented:
recheck

YanOctavian commented:
> I have read and hereby sign the Contributor License Agreement.

I have read and hereby sign the Contributor License Agreement.

> **CLA Assistant Lite bot** All contributors have signed the CLA ✍️ ✅

I have read and hereby sign the Contributor License Agreement.

> I have read and hereby sign the Contributor License Agreement.

YanOctavian commented:
> **CLA Assistant Lite bot** All contributors have signed the CLA ✍️ ✅

I have read and hereby sign the Contributor License Agreement.

YanOctavian commented:
> recheck
Done. Thank you!

YanOctavian commented:
> recheck

Done, thank you!

YanOctavian commented:
> Thanks for the application, @YanOctavian. I will try and get some feedback on this from DevRel. I will also share it with the rest of the committee. In the meantime, could you provide a registered address, either in the application or via [email](mailto:grants@web3.foundation)?

Excellent!  I have added my address to the application.

YanOctavian commented:
> > Thanks for the application, @YanOctavian. I will try and get some feedback on this from DevRel. I will also share it with the rest of the committee. In the meantime, could you provide a registered address, either in the application or via [email](mailto:grants@web3.foundation)?
> 
> Excellent! I have added my address to the application.

@semuelle Sorry to bother you, the application has been basically supplemented and updated. We would like to know what you think of this project so that we can further develop it. We are very honored to be able to provide products and ideas for the polkadot ecosystem. Looking forward to getting your replies and suggestions.

Noc2 reviewed (changes requested):
Thanks a lot for the application. I have one initial question: Why do you think there is an urgent need for sparse Merkle trees , and who is the large user base? Do you want to use it for a specific project? Do you know somebody that depends on it, etc?

YanOctavian commented:
> Thanks a lot for the application. I have one initial question: Why do you think there is an urgent need for sparse Merkle trees , and who is the large user base? Do you want to use it for a specific project? Do you know somebody that depends on it, etc?

@Noc2 thank you for your reply. Yes, we have implemented Sparse Merkle Trees in our own project. Off-chain storage technology has become quite common, and many teams have this requirement. In our project, we use Sparse Merkle Trees due to their characteristic of having a completely binary path, which is particularly friendly to zero-knowledge proof circuits. As you know, off-chain computation and storage are often combined with zero-knowledge proofs.

While using Rust in our team, we found that development in this area was lacking. Either the existing verification methods were not user-friendly, or they consumed too much gas. We spent over a month working on code writing and testing in this area, hoping to provide our final technical solution to the Polkadot ecosystem to simplify development. Moreover, in the Ethereum ecosystem, we discovered a large number of examples using Sparse Merkle Trees. All of this indicates that the Sparse Merkle Tree solution is widely adopted.

Given that the Polkadot ecosystem has the highest technical activity and the fastest-growing developer community, there are undoubtedly many potential teams using such technology. We aim to contribute to making development simpler in this ecosystem.

YanOctavian commented:
> > Thanks a lot for the application. I have one initial question: Why do you think there is an urgent need for sparse Merkle trees , and who is the large user base? Do you want to use it for a specific project? Do you know somebody that depends on it, etc?
> 
> @Noc2 thank you for your reply. Yes, we have implemented Sparse Merkle Trees in our own project. Off-chain storage technology has become quite common, and many teams have this requirement. In our project, we use Sparse Merkle Trees due to their characteristic of having a completely binary path, which is particularly friendly to zero-knowledge proof circuits. As you know, off-chain computation and storage are often combined with zero-knowledge proofs.
> 
> While using Rust in our team, we found that development in this area was lacking. Either the existing verification methods were not user-friendly, or they consumed too much gas. We spent over a month working on code writing and testing in this area, hoping to provide our final technical solution to the Polkadot ecosystem to simplify development. Moreover, in the Ethereum ecosystem, we discovered a large number of examples using Sparse Merkle Trees. All of this indicates that the Sparse Merkle Tree solution is widely adopted.
> 
> Given that the Polkadot ecosystem has the highest technical activity and the fastest-growing developer community, there are undoubtedly many potential teams using such technology. We aim to contribute to making development simpler in this ecosystem.

[https://github.com/Orbiter-Finance/OB_Submitter/blob/main/crates/off-chain-state/src/lib.rs](https://github.com/Orbiter-Finance/OB_Submitter/blob/main/crates/off-chain-state/src/lib.rs) This is the interface of our off-chain part, and the code of the upstream library has also been made necessary Change. Of course, if it is compatible with substrate, there is still similar work to be done. This is just an example used in Ethereum smart contracts. We now hope to be compatible with both ecologies at the same time

YanOctavian commented:
> > > Thanks a lot for the application. I have one initial question: Why do you think there is an urgent need for sparse Merkle trees , and who is the large user base? Do you want to use it for a specific project? Do you know somebody that depends on it, etc?
> > 
> > 
> > @Noc2 thank you for your reply. Yes, we have implemented Sparse Merkle Trees in our own project. Off-chain storage technology has become quite common, and many teams have this requirement. In our project, we use Sparse Merkle Trees due to their characteristic of having a completely binary path, which is particularly friendly to zero-knowledge proof circuits. As you know, off-chain computation and storage are often combined with zero-knowledge proofs.
> > While using Rust in our team, we found that development in this area was lacking. Either the existing verification methods were not user-friendly, or they consumed too much gas. We spent over a month working on code writing and testing in this area, hoping to provide our final technical solution to the Polkadot ecosystem to simplify development. Moreover, in the Ethereum ecosystem, we discovered a large number of examples using Sparse Merkle Trees. All of this indicates that the Sparse Merkle Tree solution is widely adopted.
> > Given that the Polkadot ecosystem has the highest technical activity and the fastest-growing developer community, there are undoubtedly many potential teams using such technology. We aim to contribute to making development simpler in this ecosystem.
> 
> https://github.com/Orbiter-Finance/OB_Submitter/blob/main/crates/off-chain-state/src/lib.rs This is the interface of our off-chain part, and the code of the upstream library has also been made necessary Change. Of course, if it is compatible with substrate, there is still similar work to be done. This is just an example used in Ethereum smart contracts. We now hope to be compatible with both ecologies at the same time

[https://github.com/nervosnetwork/sparse-merkle-tree/blob/master/src/merkle_proof.rs#L537](https://github.com/nervosnetwork/sparse-merkle-tree/blob/master/src/merkle_proof.rs#L537)Such similar verification methods are difficult to use in smart contracts. We currently use verification ideas that save gas and are easier for developers to understand.

YanOctavian commented:
> > > > Thanks a lot for the application. I have one initial question: Why do you think there is an urgent need for sparse Merkle trees , and who is the large user base? Do you want to use it for a specific project? Do you know somebody that depends on it, etc?
> > > 
> > > 
> > > @Noc2 thank you for your reply. Yes, we have implemented Sparse Merkle Trees in our own project. Off-chain storage technology has become quite common, and many teams have this requirement. In our project, we use Sparse Merkle Trees due to their characteristic of having a completely binary path, which is particularly friendly to zero-knowledge proof circuits. As you know, off-chain computation and storage are often combined with zero-knowledge proofs.
> > > While using Rust in our team, we found that development in this area was lacking. Either the existing verification methods were not user-friendly, or they consumed too much gas. We spent over a month working on code writing and testing in this area, hoping to provide our final technical solution to the Polkadot ecosystem to simplify development. Moreover, in the Ethereum ecosystem, we discovered a large number of examples using Sparse Merkle Trees. All of this indicates that the Sparse Merkle Tree solution is widely adopted.
> > > Given that the Polkadot ecosystem has the highest technical activity and the fastest-growing developer community, there are undoubtedly many potential teams using such technology. We aim to contribute to making development simpler in this ecosystem.
> > 
> > 
> > https://github.com/Orbiter-Finance/OB_Submitter/blob/main/crates/off-chain-state/src/lib.rs This is the interface of our off-chain part, and the code of the upstream library has also been made necessary Change. Of course, if it is compatible with substrate, there is still similar work to be done. This is just an example used in Ethereum smart contracts. We now hope to be compatible with both ecologies at the same time
> 
> https://github.com/nervosnetwork/sparse-merkle-tree/blob/master/src/merkle_proof.rs#L537Such similar verification methods are difficult to use in smart contracts. We currently use verification ideas that save gas and are easier for developers to understand.

Of course, our goal is not how to change these upstream libraries, but how to provide simpler sparse Merkle tree pallets and ink smart contracts for developers to use directly. And a corresponding off-chain backend

YanOctavian commented:
> > > > > Thanks a lot for the application. I have one initial question: Why do you think there is an urgent need for sparse Merkle trees , and who is the large user base? Do you want to use it for a specific project? Do you know somebody that depends on it, etc?
> > > > 
> > > > 
> > > > @Noc2 thank you for your reply. Yes, we have implemented Sparse Merkle Trees in our own project. Off-chain storage technology has become quite common, and many teams have this requirement. In our project, we use Sparse Merkle Trees due to their characteristic of having a completely binary path, which is particularly friendly to zero-knowledge proof circuits. As you know, off-chain computation and storage are often combined with zero-knowledge proofs.
> > > > While using Rust in our team, we found that development in this area was lacking. Either the existing verification methods were not user-friendly, or they consumed too much gas. We spent over a month working on code writing and testing in this area, hoping to provide our final technical solution to the Polkadot ecosystem to simplify development. Moreover, in the Ethereum ecosystem, we discovered a large number of examples using Sparse Merkle Trees. All of this indicates that the Sparse Merkle Tree solution is widely adopted.
> > > > Given that the Polkadot ecosystem has the highest technical activity and the fastest-growing developer community, there are undoubtedly many potential teams using such technology. We aim to contribute to making development simpler in this ecosystem.
> > > 
> > > 
> > > https://github.com/Orbiter-Finance/OB_Submitter/blob/main/crates/off-chain-state/src/lib.rs This is the interface of our off-chain part, and the code of the upstream library has also been made necessary Change. Of course, if it is compatible with substrate, there is still similar work to be done. This is just an example used in Ethereum smart contracts. We now hope to be compatible with both ecologies at the same time
> > 
> > 
> > https://github.com/nervosnetwork/sparse-merkle-tree/blob/master/src/merkle_proof.rs#L537Such similar verification methods are difficult to use in smart contracts. We currently use verification ideas that save gas and are easier for developers to understand.
> 
> Of course, our goal is not how to change these upstream libraries, but how to provide simpler sparse Merkle tree pallets and ink smart contracts for developers to use directly. And a corresponding off-chain backend



> > > > > Thanks a lot for the application. I have one initial question: Why do you think there is an urgent need for sparse Merkle trees , and who is the large user base? Do you want to use it for a specific project? Do you know somebody that depends on it, etc?
> > > > 
> > > > 
> > > > @Noc2 thank you for your reply. Yes, we have implemented Sparse Merkle Trees in our own project. Off-chain storage technology has become quite common, and many teams have this requirement. In our project, we use Sparse Merkle Trees due to their characteristic of having a completely binary path, which is particularly friendly to zero-knowledge proof circuits. As you know, off-chain computation and storage are often combined with zero-knowledge proofs.
> > > > While using Rust in our team, we found that development in this area was lacking. Either the existing verification methods were not user-friendly, or they consumed too much gas. We spent over a month working on code writing and testing in this area, hoping to provide our final technical solution to the Polkadot ecosystem to simplify development. Moreover, in the Ethereum ecosystem, we discovered a large number of examples using Sparse Merkle Trees. All of this indicates that the Sparse Merkle Tree solution is widely adopted.
> > > > Given that the Polkadot ecosystem has the highest technical activity and the fastest-growing developer community, there are undoubtedly many potential teams using such technology. We aim to contribute to making development simpler in this ecosystem.
> > > 
> > > 
> > > https://github.com/Orbiter-Finance/OB_Submitter/blob/main/crates/off-chain-state/src/lib.rs This is the interface of our off-chain part, and the code of the upstream library has also been made necessary Change. Of course, if it is compatible with substrate, there is still similar work to be done. This is just an example used in Ethereum smart contracts. We now hope to be compatible with both ecologies at the same time
> > 
> > 
> > https://github.com/nervosnetwork/sparse-merkle-tree/blob/master/src/merkle_proof.rs#L537Such similar verification methods are difficult to use in smart contracts. We currently use verification ideas that save gas and are easier for developers to understand.
> 
> Of course, our goal is not how to change these upstream libraries, but how to provide simpler sparse Merkle tree pallets and ink smart contracts for developers to use directly. And a corresponding off-chain backend

After early development and testing on Ethereum, we now have very complete pallet and ink smart contracts and backend implementation solutions, and we hope to provide such technical implementation for the Polkadot ecosystem. This is a basic development solution and a technology that can be reused by all teams using substrate and ink. To facilitate the development of others, this is our original intention in creating this project.

Noc2 reviewed (changes requested):
Thanks for the quick reply. I have some follow-up questions: 
- Could you tell me more about your project (it's not orbiter?) that leverages Sparse Merkle trees? 
- Do I understand it correctly that you already finished the work of this grant?

YanOctavian commented:
> Thanks for the quick reply. I have some follow-up questions:
> 
> * Could you tell me more about your project (it's not orbiter?) that leverages Sparse Merkle trees?
> * Do I understand it correctly that you already finished the work of this grant?

@Noc2 Great Job! In the orbiter team, we developed the [submitter](https://github.com/Orbiter-Finance/OB_Submitter) project using rust. This project was developed by me. It tracks the transfer of specific contracts on L2 and the Ethereum chain, and then stores the account amount in an off-chain sparse Merkle tree. Provide transaction proof and verification through zero-knowledge proof, and then provide Merkel root to the chain. We found that the functions of the sparse Merkel tree can be completely independent and used as development templates for the community to facilitate developer development. At present, we have only completed some functions in Ethereum, and are considering continued development of substrate and ink smart contracts that are compatible with Polkadot to support the technological development of the Polkadot community. We are also senior developers of the substrate ecosystem,  have more than 3 years of development experience in the Polkadot community and fully understand the needs of the community. This is the origin of this grant project.

YanOctavian commented:
> > Thanks for the quick reply. I have some follow-up questions:
> > 
> > * Could you tell me more about your project (it's not orbiter?) that leverages Sparse Merkle trees?
> > * Do I understand it correctly that you already finished the work of this grant?
> 
> @Noc2 Great Job! In the orbiter team, we developed the [submitter](https://github.com/Orbiter-Finance/OB_Submitter) project using rust. The core development of this project will be completed by me. It tracks the transfer of specific contracts on L2 and the Ethereum chain, and then stores the account amount in an off-chain sparse Merkle tree. Provide transaction proof and verification through zero-knowledge proof, and then provide Merkel root to the chain. We found that the functions of the sparse Merkel tree can be completely independent and used as development templates for the community to facilitate developer development. At present, we have only completed some functions in Ethereum, and are considering continued development of substrate and ink smart contracts that are compatible with Polkadot to support the technological development of the Polkadot community. We are also senior developers of the substrate ecosystem, have more than 3 years of development experience in the Polkadot community and fully understand the needs of the community. So this grant project was created.

@Noc2 [https://github.com/Orbiter-Finance/papers/blob/main/yellowpaper/yellowpaper.md#off-chain-component](https://github.com/Orbiter-Finance/papers/blob/main/yellowpaper/yellowpaper.md#off-chain-component)This is the yellow paper of our Orbiter team, which mentions the submitter client. We are very much looking forward to you learning about it, if you are interested.

YanOctavian commented:
> > > Thanks for the quick reply. I have some follow-up questions:
> > > 
> > > * Could you tell me more about your project (it's not orbiter?) that leverages Sparse Merkle trees?
> > > * Do I understand it correctly that you already finished the work of this grant?
> > 
> > 
> > @Noc2 Great Job! In the orbiter team, we developed the [submitter](https://github.com/Orbiter-Finance/OB_Submitter) project using rust. The core development of this project will be completed by me. It tracks the transfer of specific contracts on L2 and the Ethereum chain, and then stores the account amount in an off-chain sparse Merkle tree. Provide transaction proof and verification through zero-knowledge proof, and then provide Merkel root to the chain. We found that the functions of the sparse Merkel tree can be completely independent and used as development templates for the community to facilitate developer development. At present, we have only completed some functions in Ethereum, and are considering continued development of substrate and ink smart contracts that are compatible with Polkadot to support the technological development of the Polkadot community. We are also senior developers of the substrate ecosystem, have more than 3 years of development experience in the Polkadot community and fully understand the needs of the community. So this grant project was created.
> 
> @Noc2 https://github.com/Orbiter-Finance/papers/blob/main/yellowpaper/yellowpaper.md#off-chain-componentThis is the yellow paper of our Orbiter team, which mentions the submitter client. We are very much looking forward to you learning about it, if you are interested.

We will use a lot of substrate technology in the future, such as the decentralized sequencer of cross-chain bridges, which requires the use of substrate's transaction pool and p2p. substrate is a very excellent technical framework, thank you for your work. We also hope to provide good ideas to the Polkadot community and jointly promote the community’s technology development.

semuelle commented:
Hi @andredif, sorry for the inconvenience, but we [just started requiring KYC/KYB checks](https://grants.web3.foundation/docs/faq#what-is-kyckyb-and-why-do-i-have-to-provide-this-data) for all potential grantees. Could you please provide the information outlined under [this link](https://in.sumsub.com/idensic/l/#/uni_JRqwo72Y5AjXaRDO)? Alternatively, if the beneficiary is not an entity but individual(s) (I can't tell based on the information in the application), please follow [this link](https://in.sumsub.com/idensic/l/#/uni_mjcB05gJZfh8dbr7). Let me know if you have any questions or issues!

octavei commented:
> Hi @YanOctavian, sorry for the inconvenience, but we [just started requiring KYC/KYB checks](https://grants.web3.foundation/docs/faq#what-is-kyckyb-and-why-do-i-have-to-provide-this-data) for all potential grantees. Could you please provide the information outlined under [this link](https://in.sumsub.com/idensic/l/#/uni_JRqwo72Y5AjXaRDO)? Alternatively, if the beneficiary is not an entity but individual(s) (I can't tell based on the information in the application), please follow [this link](https://in.sumsub.com/idensic/l/#/uni_mjcB05gJZfh8dbr7). Let me know if you have any questions or issues!

@semuelle We understand your requirements very well, and have a question here:
As one of the developers of this project, I am the only one who provides KYC. Is that okay? We currently do not have an entity.

YanOctavian commented:
> > Hi @YanOctavian, sorry for the inconvenience, but we [just started requiring KYC/KYB checks](https://grants.web3.foundation/docs/faq#what-is-kyckyb-and-why-do-i-have-to-provide-this-data) for all potential grantees. Could you please provide the information outlined under [this link](https://in.sumsub.com/idensic/l/#/uni_JRqwo72Y5AjXaRDO)? Alternatively, if the beneficiary is not an entity but individual(s) (I can't tell based on the information in the application), please follow [this link](https://in.sumsub.com/idensic/l/#/uni_mjcB05gJZfh8dbr7). Let me know if you have any questions or issues!
> 
> @semuelle We understand your requirements very well, and have a question here: As one of the developers of this project, I am the only one who provides KYC. Is that okay? We currently do not have an entity.

@semuelle Does grant reject applications from certain countries? Our current team has no entity and is working in a distributed manner. I am from the Philippines and my colleague @octavei  is still in China. In the grant application, I filled in the registered address as my own. If applying for funds is not prohibited in any country or region, I should give you my colleague’s detailed address and let him do the KYC. Of course, if you need me to KYC, I could, but at the moment I have no plans to provide personal privacy information. If there is a better suggestion, we are willing to adopt it. Thanks again!

YanOctavian commented:
> > > Hi @YanOctavian, sorry for the inconvenience, but we [just started requiring KYC/KYB checks](https://grants.web3.foundation/docs/faq#what-is-kyckyb-and-why-do-i-have-to-provide-this-data) for all potential grantees. Could you please provide the information outlined under [this link](https://in.sumsub.com/idensic/l/#/uni_JRqwo72Y5AjXaRDO)? Alternatively, if the beneficiary is not an entity but individual(s) (I can't tell based on the information in the application), please follow [this link](https://in.sumsub.com/idensic/l/#/uni_mjcB05gJZfh8dbr7). Let me know if you have any questions or issues!
> 
> Hi, @semuelle. Does grant reject applications from certain countries? Our current team has no entity and is working in a distributed manner. I am from the Philippines and my colleague @octavei is still in China. In the grant application, I filled in the registered address as my own. If applying for funds is not prohibited in any country or region, I should give you my colleague’s detailed address and let him do the KYC. Of course, if you need me to KYC, I could, but at the moment I have no plans to provide personal privacy information. If there is a better suggestion, we are willing to adopt it. Thanks again!

YanOctavian commented:
> > > Hi @YanOctavian, sorry for the inconvenience, but we [just started requiring KYC/KYB checks](https://grants.web3.foundation/docs/faq#what-is-kyckyb-and-why-do-i-have-to-provide-this-data) for all potential grantees. Could you please provide the information outlined under [this link](https://in.sumsub.com/idensic/l/#/uni_JRqwo72Y5AjXaRDO)? Alternatively, if the beneficiary is not an entity but individual(s) (I can't tell based on the information in the application), please follow [this link](https://in.sumsub.com/idensic/l/#/uni_mjcB05gJZfh8dbr7). Let me know if you have any questions or issues!
> 
> Hi, @semuelle. Does grant reject applications from certain countries? Our current team has no entity and is working in a distributed manner. I am from the Philippines and my colleague @octavei is still in China. In the grant application, I filled in the registered address as my own. If applying for funds is not prohibited in any country or region, I should give you my colleague’s detailed address and let him do the KYC. Of course, if you need me to KYC, I could, but at the moment I have no plans to provide personal privacy information. If there is a better suggestion, we are willing to adopt it. Thanks again!

@semuelle You know, we have many teams in the blockchain field, and they have different needs for team privacy. As developers, we sometimes serve different teams at the same time. If we do not comply with the agreement and expose personal information, it will bring risks to them. Having said that, I still think it's reasonable to do what you asked here as it relates to tax compliance. But if you can also consider our privacy requirements, we will be very happy. For example, just ask my colleague @octavei  to KYC here.

octavei commented:
> Hi @YanOctavian & @octavei, I understand your concerns and I'm sure we'll find a way forward. We don't want to encumber anyone's work, we just have to make sure that we follow legal requirements.
> 
> First of all, unlike the data in the grant application, we are not asking you to _publish_ personal information. The data goes into our KYC provider's form and is only accessible for us for verification purposes. By sharing your data with us that way, you don't have to provide it in the public grant application anymore, which might actually help with privacy.
> 
> > Our current team has no entity and is working in a distributed manner.
> 
> I think this is fairly common in our ecosystem and shouldn't be an issue. For us as grants team, it does not matter who the beneficiary of the grant is beyond their previous experience and possibly reputation in the ecosystem. The foundation, however, needs to be able to prove that grants aren't sent to, for example, entities under sanctions. That doesn't mean that every employee needs to perform KYC, but at least the _invoicing_ person/entity.
> 
> I hope that helps. Feel free to reach out to me (@sebastian:web3.foundation) if you have any further questions.

Thank you for your understanding and explanation @semuelle  , I will KYC soon. In addition, I would like to ask another question: 
Do the registered address and contact information in the application need to be changed?  because they were @YanOctavian 's before.

semuelle commented:
> Do the registered address and contact information in the application need to be changed? because they were @YanOctavian 's before.

Ideally, yes. If you prefer not to disclose them publicly, feel free to [email them to us](mailto:grants@web3.foundation).

octavei commented:
> > Do the registered address and contact information in the application need to be changed? because they were @YanOctavian 's before.
> 
> Ideally, yes. If you prefer not to disclose them publicly, feel free to [email them to us](mailto:grants@web3.foundation).
@semuelle I have KYC and updated our application contact information. 
The email address is octavei@proton.me.

YanOctavian commented:
> Refere

@keeganquigley Thank you once again for your approval and support! Both my colleague @octavei and I truly appreciate it.

YanOctavian commented:
> Thanks for completing the KYC @YanOctavian I'm happy to go ahead with it.


@keeganquigley Thank you once again for your approval and support! Both my colleague @octavei and I truly appreciate it.

semuelle reviewed (approved):
Sorry again for the wait, @YanOctavian. I have no further questions.

keeganquigley commented:
Hi @YanOctavian how is M1 coming along?

octavei commented:
> Hi @YanOctavian how is M1 coming along?
@keeganquigley 
Happy Lunar New Year. I'm sorry that I didn't tell you in time about the temporary postponement of M1. I am very passionate about community decentralized projects. I have provided technical education and guidance on Dota for two months, and also carried out part of the development work. I am currently preparing to end it. At present, subSMT is in the final stage and M1 will be completed before the end of the month. Thanks again！

keeganquigley commented:
Hi @octavei just checking in since it is past the month's end :)

octavei commented:
> Hi @octavei just checking in since it is past the month's end :)
 I'm very sorry, I've just been busy with other things until now. Give me another two weeks and I will definitely complete the first milestone. Thanks again for the reminder. Happy life! @keeganquigley

octavei commented:
> Reference

From today on, I will go all out to complete it.

octavei commented:
> Hi @octavei just checking in since it is past the month's end :)

Sorry to bother you again. I have another question here. We have developed a [marketplace](https://dot20.market/market) for Dota, which is currently in the closed-source stage. We will support more Polkadot ecological assets and more inscriptions for transactions later. This kind of work is time-consuming and should also become a public resource for the community to avoid repeated development by the community. If we are willing to open source, will we receive grant support?

octavei commented:
> > Hi @octavei just checking in since it is past the month's end :)
> 
> Sorry to bother you again. I have another question here. We have developed a [marketplace](https://dot20.market/market) for [Dota](https://dota.fyi/inscribe/), which is currently in the closed-source stage. We will support more Polkadot ecological assets and more inscriptions for trade later. This kind of work is time-consuming and should also become a public resource for the community to avoid repeated development by the community. If we are willing to open source, will we receive grant support? @keeganquigley

octavei commented:
And this is [our telegrame](https://t.me/dot20market).  If you have more questions, you can join the group to participate in the discussion or provide more suggestions.

octavei commented:
Such a product is very meaningful. It allows Polkadot ecological assets, whether they come from parallel chains, relay chains, or independent chains, to quickly build their own market, or to trade in the dot20market we established. , Promote the circulation of Polkadot ecological assets. And this is what we are doing

octavei commented:
@semuelle @keeganquigley @Noc2

keeganquigley commented:
Hi @YanOctavian to answer your question, we don't typically support retroactive payments, and since I can't see the project since it is closed-source, it is hard to judge if it is something we'd be interested in, but since it has already been built it probably wouldn't be eligible for a grant, unfortunately. But I appreciate the update and sounds good, looking forward to the delivery.

keeganquigley commented:
Hi @YanOctavian how is the delivery coming along?

keeganquigley commented:
pinging @YanOctavian

keeganquigley commented:
Hi @YanOctavian as we haven't heard from you in a couple months, please let us know if you are still planning to submit a delivery soon for this milestone. If it is significantly delayed, we would ask for an amendment to be submitted to extend the timeline. Thanks!

octavei commented:
I apologize for the delayed response. We have indeed faced some
difficulties and challenges over the past two months, which we are
currently working to resolve. subSMT is a fantastic idea and project, and
we do intend to incorporate it into our upcoming projects. I would like to
request an extension to complete it by next month. In reality, most of the
development work has already been done, but we need time to update the code
and write documentation, which we currently lack. We plan to fully dedicate
ourselves to completing the first milestone starting from the 1st of next
month.

Keegan | W3F ***@***.***> 于2024年5月3日周五 23:46写道：

> Hi @YanOctavian <https://github.com/YanOctavian> as we haven't heard from
> you in a couple months, please let us know if you are still planning to
> submit a delivery soon for this milestone. If it is significantly delayed,
> we would ask for an amendment to be submitted to extend the timeline.
> Thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grants-Program/pull/2104#issuecomment-2093274189>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4WHBIHHJK6UABZUFAQNPCLZAOWOJAVCNFSM6AAAAAA7IBFIOKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAOJTGI3TIMJYHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

keeganquigley commented:
Hi @YanOctavian thanks for the update.

Sure thing, feel free to amend the original contract to extend the timeline, and then create a PR for it at https://github.com/w3f/Grants-Program

This should be approved relatively quickly since it's a simple update. Thanks!

keeganquigley commented:
Hi @YanOctavian if delivered in the next 2 - 3 weeks probably no amendment needed. But any longer than that and the above would apply. Let me know if you still plan to submit this. Thanks.

keeganquigley commented:
Hi @YanOctavian just checking if you are able to submit an amendment for this? You can extend it for 2 months if necessary and that will take off any added pressure. Thanks.

keeganquigley commented:
Hi @YanOctavian at this point I'd say let's go ahead and cancel it and you can potentially pick it back up in the future should you want to work on it. Let me know if this sounds good, otherwise we'll close it in a few days. Thanks.

YanOctavian commented:
> 你好@YanOctavian此时，我想说我们先取消它，以后如果你想继续做的话，你可以重新开始。如果这听起来不错，请告诉我，否则我们会在几天内关闭它。谢谢。

Thank you for your reply and I apologize for being so late in replying to you. I've been really burned out on some personal team stuff lately. But I still prioritize this project, please give me some time. If I can, I will temporarily stop other things and try to finish it first. I agree with the idea of ​​extending it by 2 months, which would relieve me of the extra pressure. Thank you again for your professionalism and continued follow-up with us

YanOctavian commented:
> Hi @YanOctavian at this point I'd say let's go ahead and cancel it and you can potentially pick it back up in the future should you want to work on it. Let me know if this sounds good, otherwise we'll close it in a few days. Thanks.

Thank you for your reply and I apologize for being so late in replying to you. I've been really burned out on some personal team stuff lately. But I still prioritize this project, please give me some time. If I can, I will temporarily stop other things and try to finish it first. I agree with the idea of ​​extending it by 2 months, which would relieve me of the extra pressure. Thank you again for your professionalism and continued follow-up with us

YanOctavian commented:
> Hi @YanOctavian if delivered in the next 2 - 3 weeks probably no amendment needed. But any longer than that and the above would apply. Let me know if you still plan to submit this. Thanks.

In addition, I have a question. If the deadline is extended, which part needs to be updated by submitting PR?

semuelle commented:
> If the deadline is extended, which part needs to be updated by submitting PR?

Hi @YanOctavian. Under `Development Roadmap`, there are declarations of total duration and milestone durations. You can simply change these to a more realistic timeframe.

YanOctavian commented:
> > If the deadline is extended, which part needs to be updated by submitting PR?
> 
> Hi @YanOctavian. Under `Development Roadmap`, there are declarations of total duration and milestone durations. You can simply change these to a more realistic timeframe.

https://github.com/w3f/Grants-Program/pull/2335 , @semuelle Thank you very much for your reminder. I have submitted a new PR and changed the milestone time to a specific time range.

keeganquigley commented:
Hi @YanOctavian just checking in to see how milestone 1 is coming along.

keeganquigley commented:
Hi @YanOctavian seeing as M1 was due about a month ago under the new amendment, just seeing how progress is coming along. Thanks for any update you can provide.

octavei commented:
Thank you very much for your letter. I have just joined the company in the
past two months. These two months have been really busy for me. I have
helped two teams use substrate to develop public chains. I only have time
to reply to you now. I hope to complete this milestone next October. I will
have plenty of personal time in October. Thanks again for your understanding

Keegan | W3F ***@***.***> 于2024年8月30日周五 02:54写道：

> Hi @YanOctavian <https://github.com/YanOctavian> seeing as M1 was due
> about a month ago under the new amendment, just seeing how progress is
> coming along. Thanks for any update you can provide.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grants-Program/pull/2104#issuecomment-2318630997>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4WHBIFQT3HP7SARDKUDQYDZT5U43AVCNFSM6AAAAAA7IBFIOKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMJYGYZTAOJZG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

keeganquigley commented:
Hi @octavei still planning to deliver soon?

keeganquigley commented:
pinging @YanOctavian

octavei commented:
Thank you very much for your reply. We are currently in the testing phase
and expect to complete it by the 25th of this month. At the same time, we
are very pleased to have found a team in the community that is using this
project. They are participating in the final testing with us, and we will
benefit from their valuable feedback.

Keegan | W3F ***@***.***> 于2024年11月8日周五 05:43写道：

> pinging @YanOctavian <https://github.com/YanOctavian>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grants-Program/pull/2104#issuecomment-2463284071>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4WHBIG7GK7A5DJSRH7I733Z7PNIJAVCNFSM6AAAAAA7IBFIOKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINRTGI4DIMBXGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

keeganquigley commented:
Sounds good thanks @YanOctavian if you find that the delivery will be delayed past November, please consider filing an amendment to extend the timeline. Thanks and have a great weekend :)

keeganquigley commented:
hi @YanOctavian just checking in :)

octavei commented:
good day!

Keegan | W3F ***@***.***> 于2024年12月4日周三 01:13写道：

> hi @YanOctavian <https://github.com/YanOctavian> just checking in :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grants-Program/pull/2104#issuecomment-2515137952>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4WHBIFZYYKLQZ7FVNSWKTT2DXREDAVCNFSM6AAAAAA7IBFIOKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMJVGEZTOOJVGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

keeganquigley commented:
@octavei any updates? If the project will be further delayed could you please consider filing an amendment to extend the timeline?

YanOctavian commented:
> @octavei any updates? If the project will be further delayed could you please consider filing an amendment to extend the timeline?

Apologies for the late reply. I've been busy with the development of this project last month. I've already submitted and had my PR for the first milestone merged, and will submit the second milestone in the next few days. I'm not sure if any updates are needed in the application. Have a great day! @keeganquigley

YanOctavian commented:
https://github.com/w3f/Grant-Milestone-Delivery/pull/1228

octavei commented:
Hi, I submitted the invoice for the first milestone on the 4th and would
like to confirm if you have received it.

yanoctavian ***@***.***> 于2024年12月5日周四 10:18写道：

> w3f/Grant-Milestone-Delivery#1228
> <https://github.com/w3f/Grant-Milestone-Delivery/pull/1228>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grants-Program/pull/2104#issuecomment-2518941735>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4WHBIBBLPC2WFDC3XSHRBL2D6ZYHAVCNFSM6AAAAAA7IBFIOKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMJYHE2DCNZTGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

semuelle commented:
@octavei, I can confirm that we have received the invoice. It should be processed fairly soon.

octavei commented:
Great, looking forward to further feedback! Thanks again!

Sebastian Müller ***@***.***> 于2024年12月17日周二 18:01写道：

> @octavei <https://github.com/octavei>, I can confirm that we have
> received the invoice. It should be processed fairly soon.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grants-Program/pull/2104#issuecomment-2548006796>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4WHBIDSC7M4MYZVE3YARSL2F7ZA7AVCNFSM6AAAAAA7IBFIOKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNBYGAYDMNZZGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

YanOctavian commented:
@semuelle @keeganquigley Just wanted to know if the M1 payment could be processed before the 25th. If so, we would be very happy. Wishing you a wonderful holiday!

keeganquigley commented:
Hi @YanOctavian just checking in to see how M2 is progressing?

keeganquigley commented:
pinging @YanOctavian

keeganquigley commented:
Hi @YanOctavian are you still working on this grant? Please let us know, otherwise we will close it in a few weeks due to inactivity.

YanOctavian commented:
> Hi @YanOctavian are you still working on this grant? Please let us know, otherwise we will close it in a few weeks due to inactivity.

@keeganquigley  Sorry for the late reply, I will submit it next week

YanOctavian commented:
@keeganquigley @semuelle @Noc2  I've already submitted [subsmt m2](https://github.com/w3f/Grant-Milestone-Delivery/pull/1254). If there are any issues, please let me know. Thank you very much.
